task1Daniv/stringlist.cpp

```diff
@@ -0,0 +1,250 @@
+#include "stringlist.h"
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+
+const int dataIndex = 0;
+const int nextNodeIndex = 1;
+
+List initialization()
+{
+    List temp = static_cast<char**>(malloc(sizeof(List) * 2));
+
+    if(temp != nullptr)
+    {
+        temp[dataIndex] = nullptr;
+        temp[nextNodeIndex] = nullptr;
+    }
+    return  temp;
+}
+
+void push(List &list, const String &string)
+{
+    if(list == nullptr)
+    {
+        return;
+    }
+    String textStr = static_cast<String>(malloc(sizeof(char) * strlen(string) + 1));
+    strcpy(textStr, string);
+
+    if(list[dataIndex] == nullptr)
+    {
+        list[dataIndex] = textStr;
+    }
+    else
+    {
+        auto temp = list;
+        while(list[nextNodeIndex] != nullptr)
+        {
+            list = reinterpret_cast<List>(list[nextNodeIndex]);
+        }
+        List node = static_cast<List>(malloc(sizeof(char*) * 2));
+        node[dataIndex] = textStr;
+        node[nextNodeIndex] = nullptr;
+        list[nextNodeIndex] = reinterpret_cast<String>(node);
+        list = temp;
+    }
+}
+
+void print(const List &list)
```

Comment:
Using **const reference** is redundant here since the list value consists of 4 bytes